src/migration-scripts/ipoe-server/1-to-2

```diff
@@ -39,12 +44,39 @@ with open(file_name, 'r') as f:
 
 config = ConfigTree(config_file)
 base = ['service', 'ipoe-server']
-pool_base = base + ['client-ip-pool']
+
 if not config.exists(base):
     exit(0)
 
+if config.exists(base + ['authentication', 'interface']):
+    for interface in config.list_nodes(base + ['authentication', 'interface']):
+        config.rename(base + ['authentication', 'interface', interface, 'mac-address'], 'mac')
+
+        mac_base = base + ['authentication', 'interface', interface, 'mac']
+        for mac in config.list_nodes(mac_base):
+            vlan_config = mac_base + [mac, 'vlan-id']
+            if config.exists(vlan_config):
+                config.rename(vlan_config, 'vlan')
+
+for interface in config.list_nodes(base + ['interface']):
+    base_path = base + ['interface', interface]
+    for vlan in ['vlan-id', 'vlan-range']:
+        if config.exists(base_path + [vlan]):
+            print(interface, vlan)
+            for tmp in config.return_values(base_path + [vlan]):
+                config.set(base_path + ['vlan'], value=tmp, replace=False)
+            config.delete(base_path + [vlan])
+
+    if config.exists(base_path + ['network-mode']):
+        tmp = config.return_value(base_path + ['network-mode'])
+        config.delete(base_path + ['network-mode'])
+        # Change L2|L3 to lower case l2|l3
+        config.set(base_path + ['mode'], value=tmp.lower())
+
+pool_base = base + ['client-ip-pool']
 if not config.exists(pool_base):
     exit(0)
```

Comment:
This is a problem if the above section is applicable but exits here before writing the modified config.

Reply:
fixed